include/Language/Haskell/Liquid/List.hs

```diff
@@ -1,5 +1,6 @@
 module Language.Haskell.Liquid.List (transpose) where
 
+{-@ lazy transpose @-}
```

Comment:
Intuitively here it should be possible to give a proper refinement type, for using `lazy` was, well, the quickest way to make this pass, hehe.